cql3/selection/selection.cc

```diff
@@ -395,14 +395,17 @@ bool result_set_builder::restrictions_filter::do_filter(const selection& selecti
     }
 
     auto static_row_iterator = static_row.iterator();
-    auto row_iterator = row.iterator();
+    auto row_iterator = row ? std::optional<query::result_row_view::iterator_type>(row->iterator()) : std::nullopt;
     auto non_pk_restrictions_map = _restrictions->get_non_pk_restriction();
     for (auto&& cdef : selection.get_columns()) {
         switch (cdef->kind) {
         case column_kind::static_column:
             // fallthrough
         case column_kind::regular_column: {
-            auto& cell_iterator = (cdef->kind == column_kind::static_column) ? static_row_iterator : row_iterator;
+            if (cdef->kind == column_kind::regular_column && !row_iterator) {
```

Comment:
Nitpick: If you just switched the two "case"s (static_column and regular_column), and which falls through to which, you wouldn't need to check cdef-kind here again. The if(!row_iterator) could be just in the regular_column case.

Reply:
Yeah - it was a concious decision. I felt tempted to simply switch the cases, but I think that it will make the code much less backport-friendly and could easily lead to incorrect backports, so I went with this slightly longer way. I believe the compiler will take care of generating optimal code from this one as well.